### PR TITLE
Use a Bundle Fragment to load the ConfigProviderResolver Impl

### PR DESCRIPTION
**Pull request by tevans78**
This is just a prototype to see if we can avoid using setInstance, do not merge
#build

### CODE REVIEW

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.microprofile.config.1.4.fragment/src/com/ibm/ws/microprofile/config14/fragment/Config14ProviderResolverOSGiStub.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    /**
+     * Get the OSGi bundle context for the given class
+     *
+     * @param clazz the class to find the context for
+     * @return the bundle context
+     */
+    static BundleContext getBundleContext() {
+        BundleContext context = null; //we'll return null if not running inside an OSGi framework (e.g. unit test) or framework is shutting down
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.config.1.4.fragment/src/com/ibm/ws/microprofile/config14/fragment/Config14ProviderResolverOSGiStub.java:64]: Probably better not to do this.

Elsewhere in this class, you assume that the result of `getBundleContext()` is not null.

File: dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/impl/AbstractProviderResolver.java [hunk 2]
```diff
@@ -74,14 +74,18 @@ protected void unsetScheduledExecutorService(ServiceReference<ScheduledExecutorS
         this.scheduledExecutorServiceRef.unsetReference(ref);
     }
 
+    protected void setStaticInstance(ConfigProviderResolver instance) {
+        ConfigProviderResolver.setInstance(instance);
```
Comment (Emily-Jiang) [inline dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/impl/AbstractProviderResolver.java:78]: This method will be deleted though.

Comment (Emily-Jiang) [reply dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/impl/AbstractProviderResolver.java:78]: ah. This is for 1.1 version. Ignore the above comment.

**Comment by Emily-Jiang** [comment]
@tevans78 The changes look good.

File: dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14ProviderResolverImpl.java [hunk 3]
```diff
@@ -20,4 +22,9 @@
     protected AbstractConfigBuilder newBuilder(ClassLoader classLoader) {
         return new Config14BuilderImpl(classLoader, getScheduledExecutorService(), getInternalConfigSources());
     }
+
+    @Override
+    protected void setStaticInstance(ConfigProviderResolver instance) {
+        //NO-OP ... this version relies purely on the ServiceLoader mechanism
+    }
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/Config14ProviderResolverImpl.java:29]: This is picky, but I don't like having a method that says that it does something, and then is overridden to not do that thing.

I don't think it's important for this prototype, but I think I'd prefer a more vague name for the method.

**Pull request closed without merging**